routemaster/mixins/log.rb

```diff
@@ -8,15 +8,7 @@ module Log
       protected
 
       def _log
-        @@_logger ||= begin
-          file_path = ENV['ROUTEMASTER_LOG_FILE']
-          file = (file_path && File.exist?(file_path)) ? File.open(file_path, 'a') : $stderr
-          level = Logger.const_get(ENV.fetch('ROUTEMASTER_LOG_LEVEL', 'INFO'))
-          Logger.new(file).tap do |logger|
-            logger.level     = level
-            logger.formatter = method(:_formatter)
-          end
-        end
+        @@_logger ||= Services::Logger.new
```

Comment:
Yep, it's not thread safe. But It's going to be initialised only once before the threads start their lives and also as you said since it's not holding any state so no prob if it's initialised twice.